ci/scripts/setup-adapters.sh

```diff
@@ -0,0 +1,103 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SCRIPTDIR=$(cd $(dirname $0); pwd)
+source $SCRIPTDIR/../../thirdparty/velox/scripts/setup-helper-functions.sh
+# Propagate errors and improve debugging.
+set -eufx -o pipefail
+
+DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}
+
+function install_aws-sdk-cpp {
+  local AWS_REPO_NAME="aws/aws-sdk-cpp"
+  local AWS_SDK_VERSION="1.9.96"
+
+  github_checkout $AWS_REPO_NAME $AWS_SDK_VERSION --depth 1 --recurse-submodules
+  cmake_install -DCMAKE_BUILD_TYPE=Debug -DBUILD_SHARED_LIBS:BOOL=OFF -DMINIMIZE_SIZE:BOOL=ON -DENABLE_TESTING:BOOL=OFF -DBUILD_ONLY:STRING="s3;identity-management" -DCMAKE_INSTALL_PREFIX="${DEPENDENCY_DIR}/install"
+}
+
+function install_libhdfs3 {
+  github_checkout apache/hawq master
+  cd $DEPENDENCY_DIR/hawq/depends/libhdfs3
+  if [[ "$OSTYPE" == darwin* ]]; then
+     sed -i '' -e "/FIND_PACKAGE(GoogleTest REQUIRED)/d" ./CMakeLists.txt
+     sed -i '' -e "s/dumpversion/dumpfullversion/" ./CMakeLists.txt
+  fi
+
+  if [[ "$OSTYPE" == linux-gnu* ]]; then
+    sed -i "/FIND_PACKAGE(GoogleTest REQUIRED)/d" ./CMakeLists.txt
+    sed -i "s/dumpversion/dumpfullversion/" ./CMake/Platform.cmake
+  fi
+  
+
+  local NAME=$(basename "$(pwd)")
+  local BINARY_DIR=_build
+  if [ -d "${BINARY_DIR}" ] && prompt "Do you want to rebuild ${NAME}?"; then
+    rm -rf "${BINARY_DIR}"
+  fi
+  mkdir -p "${BINARY_DIR}"
+  CPU_TARGET="${CPU_TARGET:-avx}"
+  COMPILER_FLAGS=$(get_cxx_flags $CPU_TARGET)
+
+  # CMAKE_POSITION_INDEPENDENT_CODE is required so that Velox can be built into dynamic libraries \
+  cmake -Wno-dev -B"${BINARY_DIR}" \
+    -GNinja \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+    -DCMAKE_CXX_STANDARD=17 \
+    "${INSTALL_PREFIX+-DCMAKE_PREFIX_PATH=}${INSTALL_PREFIX-}" \
+    "${INSTALL_PREFIX+-DCMAKE_INSTALL_PREFIX=}${INSTALL_PREFIX-}" \
+    -DCMAKE_CXX_FLAGS="$COMPILER_FLAGS" \
+    -DBUILD_TESTING=OFF \
+    -DKERBEROS_INCLUDE_DIRS=${KERBEROS_INCLUDE_DIRS} \
+    "$@"
+  ninja -C "${BINARY_DIR}" install
+}
+
+function install_protobuf {
+  wget https://github.com/protocolbuffers/protobuf/releases/download/v21.4/protobuf-all-21.4.tar.gz
+  tar -xzf protobuf-all-21.4.tar.gz --no-same-owner
+  cd protobuf-21.4
+  ./configure --prefix=/usr
+  make "-j$(nproc)"
+  make install
+  ldconfig
+}
+
+DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}
+cd "${DEPENDENCY_DIR}" || exit
+# aws-sdk-cpp missing dependencies
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+   # /etc/os-release is a standard way to query various distribution
+   # information and is available everywhere
+   LINUX_DISTRIBUTION=$(. /etc/os-release && echo ${ID})
+   if [[ "$LINUX_DISTRIBUTION" == "ubuntu" ]]; then
+      apt install -y --no-install-recommends libxml2-dev libgsasl7-dev uuid-dev
+   else # Assume Fedora/CentOS
+      yum -y install libxml2-devel libgsasl-devel libuuid-devel
+   fi
+fi
+
+if [[ "$OSTYPE" == darwin* ]]; then
```

Comment:
do we need consider OS besides Linux？

Reply:
No. This script is modified based on this one (https://github.com/Intel-bigdata/velox/blob/BDTK/scripts/setup-adapters.sh) to support Ubuntu 20.04. So it may contain some redundant code.